### PR TITLE
Miscellaneous updates

### PR DESCRIPTION
**Pull request by JP-Globality**
**What?**

- Capitalised request context keys in logs e.g `{"x-request-id": "1234"} -> {"X-Request-Id": "1234"}`
- Added tests for crud routes and for audit logging
- Fixing hostname extraction (i.e only swap http for https when we don't have https already in the url)
- Test cases for the hostname extraction
- Removing nosetest config as no longer needed
- Getting tests running again - updated entrypoint.sh to use pytest along with installing the required test dependencies

TODO: Run linting on codebase once reviews are done.

X-Request-Id now appearing in the logs:
image

### CODE REVIEW

File: microcosm_fastapi/tests/helpers.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class BaseFixture():
```
[the remark this answers is not in the document]
Comment (tonybajan-globality) [reply microcosm_fastapi/tests/helpers.py:1]: though I'm not massively clear why you need this, it's close to being a stdlib `SimpleNamespace`?

**Review by tonybajan-globality — APPROVED** [review]

File: microcosm_fastapi/context.py [hunk 2]
```diff
@@ -6,6 +8,15 @@
 HEADER_PREFIXES = [X_REQUEST]
 
 
+def capitalise_context(dct: Dict):
```
[the remark this answers is not in the document]
Comment (JP-Globality) [reply microcosm_fastapi/context.py]: Thanks for the typehint suggestion. Have implemented.

This function is actually only being used inside of the audit logs just to keep the capitalisation the same as how microcosm-flask outputs them 👍

File: microcosm_fastapi/namespaces.py [hunk 3]
```diff
@@ -67,7 +67,8 @@ def extract_hostname_from_request(self, request: Request) -> str:
         # If the microcosm_environment has been set, assume that we're using https
         environment = environ.get("MICROCOSM_ENVIRONMENT", None)
         if environment:
-            host_name = host_name.replace("http", "https")
+            if "https://" not in host_name and "http://" in host_name:
+                host_name = host_name.replace("http", "https")
```
[the remark this answers is not in the document]
Comment (JP-Globality) [reply microcosm_fastapi/namespaces.py:71]: True - maybe we don't need this at all. Have removed it! Will test an environment before next deployment.

File: microcosm_fastapi/tests/helpers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JP-Globality) [reply microcosm_fastapi/tests/helpers.py]: Good point - have implemented the `SimpleNamespace` as a straight swap. I didn't know about this class until now. Thanks for the suggestion.

**Review by german-glob — APPROVED** [review]

**Pull request merged**